### ISSUE TITLE
Hello world maybe an unfortunate name for learning purposes?

### ISSUE THREAD

**Issue by vdespa** [joomla/Joomla-3.2-Hello-World-Component#6]
I remember the times when I was starting learning Joomla. The Hello World tutorial, while partially useful, it always caused confusion because of MVC components which are named like `HelloWorldModelHelloWorld` are really bad examples of the naming conventions in Joomla and don't much help to get a better grasp of them.

Maybe when a general revision is to take place, implementing something closer to the real world and real needs would be beneficial for newcomers.

Like: `com_company`, `CompanyModelEmployee`.

**Comment by betweenbrain** [comment]
I completely agree, it can be very confusing. Maybe when we get the 3.x version finalized, we can look at better naming for it. `com_company` may be a good way to go.

**Comment by ghost** [comment]
Whilst I recognise this post was created a long time ago, I would have to agree with Vdespa also. I am trying to get my head round the Joomla Tutorial with the latest version 3.6.5 but the hello world doesn't work for me unfortunately. From looking at many books, videos and diagrams I've finally got my head round it but I had to find resources which didn't use the typical "hello world" as the example. I struggle with naming conventions which aren't related to what the "thing" is meant to be doing. So the same is true with trying to understand when functions say $k = $k etc. Literal thinker requires literal conventions in my case.

**Comment by stanblesk** [comment]
On the later steps appears **greeting**. It would be fine to use `HelloWorldModelGreeting`, `HelloWorldModelGreetings`, `HelloWorldViewGreeting`, `HelloWorldViewGreetings` and so on...
